lib/ansible/plugins/connection/ssh.py

```diff
@@ -268,6 +268,16 @@
         vars:
           - name: ansible_ssh_use_tty
             version_added: '2.7'
+        yaml: {key: connection.usetty}
+      pkcs11_provider:
```

Comment:
I went with option a.  If you specify the pkcs11 provider then pkcs11 is enabled, if you don't specify the provider then its disabled. I could detect if pkcs11 provider was used but no password was provided, I could print an error that a password/pin is required.  I am trying to avoid adding too many variables, it seems that if you specify a pkcs11_provider then its safe to assume your using pkcs11.  Let me know what you think.